addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java

```diff
@@ -228,14 +228,16 @@ public void sendReadMsg() {
                 @Override
                 public void log(String format) {
```

Comment:
Ideally you should just inline it and make it explicit that you do a conditional trace logging, what is the goal of doing it like this? How often do you call it?

Reply:
I had a look and it's called from the enum `NibeHeatPumpProtocolStates` which would require a static logger which is also against the code style.

Reply:
No its not preferred that is something different than not allowed.

Note that the static logger is a warning and the custom logging strings are an error.

See also:

> As we are in a dynamic OSGi environment, loggers should be non-static, when ever possible and have the name logger.

Reply:
Then a fix could be to just remove all log methods from `NibeHeatPumpProtocolContext` and use a static logger in `NibeHeatPumpProtocolStates` instead. :-)

Reply:
:like:

Reply:
I was using static logger originally, but that was asked to change during code review ;-)

Reply:
You are right, the previous message which I used was to strong, I will use message which matches the coding guidelines from now one, sorry :-)